6th HTML/CSS Project: Building with Responsive Design

nikoescobal opened the pull request:
## TNW Clone: Building with Responsive Design

In this project we:
- Replicated The Next Web website, putting emphasis on how it behaves differently depending on the size of the screen where the website is rendered. 
- It utilizes flex, grid, and positioning, as well as media queries and fluid/responsive web design principles. 
- This document validates as CSS level 3 + SVG by W3C CSS Validator and is validated by Jigsaw for HTML!

Thank you!

~KartheeK and Nik :heart:

bertil291utn commented on index.html:
```diff
@@ -0,0 +1,1561 @@
+        <p>
+          &nbsp;
+          <a href="http://jigsaw.w3.org/css-validator/check/referer">
+            <img
+              style="border: 0; width: 88px; height: 31px;"
+              src="http://jigsaw.w3.org/css-validator/images/vcss-blue"
```
## No inline styling

Using Inline Styles is probably one of the worst CSS practices that many web designers and developers are still doing to this day. It transgresses the main goal of why we use CSS – which is to separate style from the HTML structure. This breaks the specificity of web pages

- [ ] Remove all `style` inline you have in the HTML file

**Note:** Remove all inline style rules in the entire HTML page

###### _[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  border: none;
+}
+
+.footer-links-menu {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  list-style: none;
+  margin: 20px 6px;
+}
+
+.footer-copyright {
+  display: flex;
+  justify-content: center;
```
## Using the Same Rule Repetitively[1]
- You already have a display flex property in .flex class, please use it
- [ ] Create a class once and the use in each HTML tag where you need it.
- [ ] Delete all your `display:flex;` properties in entire CSS file

**Example:**


**Html file**

```HTML
<div class="flex">
```
**Note:** Do this change to all properties in this file 

<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+.latest-news h2 a {
+  text-decoration: none;
+  font-family: "Avalon", serif;
+  font-size: 28px;
+  color: #f42;
+}
+
+.card-1-mobile-text h3 a {
+  color: black;
+  text-decoration: none;
+}
+
+.latest-funding h2 a {
+  text-decoration: none;
+  font-size: 28px;
+  color: #f42;
```
## Using the Same Rule Repetitively[1]
- [ ] Create a class once and the use in each HTML tag where you need it.
- [ ] Delete all your `color: #f42;` properties in entire CSS file
- [ ] Add your HTML tags

**Example:**

**CSS file**

```CSS
.color-x {
  color: #f42;
}
```
**Html file**

```HTML
<div class="color-x">
```
**Note:** Do this change to all properties in this file 

<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  color: white;
+}
+
+.card-3 li a:hover {
+  color: black;
+}
+
+.latest-funding h2 a:hover {
+  color: black;
+}
+
+.latest-news h2 a:hover {
+  color: black;
+}
```
## Using the Same Rule Repetitively[1]
- [ ] Create a class once and the use in each HTML tag where you need it.
- [ ] Delete all your `color:black;` properties in entire CSS file
- [ ] Add your HTML tags

**Example:**

**CSS file**

```CSS
.black-color {
  color:black;
}
```
**Html file**

```HTML
<div class="black-color ">
```
**Note:** Do this change to all properties in this file 

<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  font-family: "Shentox", serif;
+}
+
+.latest-news h2 a {
+  text-decoration: none;
+  font-family: "Avalon", serif;
+  font-size: 28px;
+  color: #f42;
+}
+
+.card-1-mobile-text h3 a {
+  color: black;
+  text-decoration: none;
+}
+
+.latest-funding h2 a {
```
## Using Inefficient CSS Selectors[1]

When you are using selectors, the longer it is, the higher specificity it will comprise. This can break the CSS cascade and mess up the styles of your site.

This can be avoided by not using IDs within the selectors as they are directly specific.

 - [ ]  Please add a class to `<a>` element` and delete the left unnecessary rules

**Example:**
```CSS
/* before */
.card-1-mobile-text h3 a {

/* after */
.anchortag-class {
```
<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  font-family: "Avalon", sans-serif;
+}
+
+.card-3 li a {
+  color: #555;
+  font-size: 13px;
+}
+
+.latest-deals h2 a {
+  font-family: "Avalon", sans-serif;
+  font-size: 28px;
+  text-decoration: none;
+  color: #f42;
+}
+
+.footer-links-social li a {
```
## Using Inefficient CSS Selectors[1]

When you are using selectors, the longer it is, the higher specificity it will comprise. This can break the CSS cascade and mess up the styles of your site.

This can be avoided by not using IDs within the selectors as they are directly specific.

 - [ ]  Please add a class to `<a>` element` and delete the left unnecessary rules

**Example:**
```CSS
/* before */
.footer-links-social li a {

.card-3 li a {

.latest-deals h2 a {


/* after */
.anchortag-class {
```
<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  margin-left: 12px;
+}
+
+#first-nav ul li a {
+  font-family: "ARS Maquette", sans-serif;
+  color: #fff9;
```
.card-3 li a {## Using Inefficient CSS Selectors[1]

When you are using selectors, the longer it is, the higher specificity it will comprise. This can break the CSS cascade and mess up the styles of your site.

This can be avoided by not using IDs within the selectors as they are directly specific.

 - [ ]  Please add a class to `<a>` element` and delete the left unnecessary rules

**Example:**
```CSS
/* before */
#first-nav ul li a {


/* after */
.anchortag-class {
```
<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  color: #131313;
+}
+
+#first-nav ul:first-child li:first-child a {
+  color: white;
```
.card-3 li a {## Using Inefficient CSS Selectors[1]

When you are using selectors, the longer it is, the higher specificity it will comprise. This can break the CSS cascade and mess up the styles of your site.

This can be avoided by not using IDs within the selectors as they are directly specific.

 - [ ]  Please add a class to `<a>` element` and delete the left unnecessary rules

**Example:**
```CSS
/* before */
#first-nav ul:first-child li:first-child a {

/* after */
.anchortag-class {
```
<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

bertil291utn commented on style.css:
```diff
@@ -0,0 +1,1498 @@
+  right: 0;
+}
+
+.card-3-bottom div h4 {
+  font-family: "ARS Maquette-bold", serif;
+  font-size: 14px;
+}
+
```
## Using Inefficient CSS Selectors[1]
When you are using selectors, the longer it is, the higher specificity it will comprise. This can break the CSS cascade and mess up the styles of your site.

This can be avoided by not using IDs within the selectors as they are directly specific.

 - [ ]  Please add a class to `<a>` element` and delete the left unnecessary rules

**Example:**
```CSS
/* before */
.card-3-bottom div h4 {
/* after */
.h4-class {
```
<hr>

_[1] [Good and bad  CSS practices](https://speckyboy.com/good-bad-css-practices/)_

nikoescobal commented:
**Status:** Changes worked on. Now awaiting review :sweat_smile:  :shipit:? Fingers crossed! :pray:

# Hey @bertil291utn thank you for this review and for the encouraging words. We definitely enjoyed coding this project.

### We're proud of our work and put a lot of time and detail to build this project with excellence. 

As you can see below, we've added changes to our project based on the requirements:

### General Requirements Addressed

:white_check_mark: Removed `.vscode`  and `.DS_STORE` directories form our project

### Specific req. 

1.  Media queries and layout issues: FIXED [:heavy_check_mark:]

- To clarify, the breakpoints specified are incorrect. The correct breakpoints are 1023px and 767px. With these breakpoints, there are no layout issues, as you previously pointed out. Please refer to the images below as proof.

| TNW WEB PAGE AT 1024px = NOT TABLET VIEW :x:  | TNW WEB PAGE VIEW AT 1023px = TABLET VIEW :white_check_mark: |
| :-------------: | :---: |
| <img width="100%" alt="tnw-site-at-1024px" src="https://user-images.githubusercontent.com/62937819/90026284-af043280-dce9-11ea-91e1-027ad7a196e6.png"> | <img width="100%" alt="tnw-site-at-1023px" src="https://user-images.githubusercontent.com/62937819/90028379-305cc480-dcec-11ea-84b2-3fa557ba96de.png"> | 
| **TNW WEB PAGE AT 768px = NOT MOBILE VIEW :x:** | **TNW WEB PAGE AT 767px = MOBILE VIEW** :white_check_mark:  |
| <img width="100%" alt="tnw-site-at-768px" src="https://user-images.githubusercontent.com/62937819/90028356-2b981080-dcec-11ea-833f-c2a614650999.png"> | <img width="100%" alt="tnw-site-at-767px" src="https://user-images.githubusercontent.com/62937819/90028334-23d86c00-dcec-11ea-89be-2799af475b6b.png"> |

| **OUR WEB PAGE AT 1023px** = TABLET VIEW :white_check_mark:  | **OUR WEB PAGE AT 767px** = MOBILE VIEW :white_check_mark:  |
| :-------------: | :---: |
| <img width="100%" alt="our-site-at-1023px" src="https://user-images.githubusercontent.com/62937819/90028270-0c00e800-dcec-11ea-8743-fda3ef0848be.png"> | <img width="100%" alt="our-site-at-767px" src="https://user-images.githubusercontent.com/62937819/90028183-ef64b000-dceb-11ea-905b-608a26e756bb.png"> |

2. Remove about links w/ medium screens approach: FIXED [:heavy_check_mark:]  

<img width="1440" alt="about-section-hidden" src="https://user-images.githubusercontent.com/62937819/90030824-1670b100-dcef-11ea-9f4c-575afb9cf7b6.png">

3. TNW logo is not in there with medium screen approach: FIXED [:heavy_check_mark:] 

<img width="1434" alt="tnw-logo-hidden" src="https://user-images.githubusercontent.com/62937819/90028318-1f13b800-dcec-11ea-8eb0-c319a65b0a3b.png">

4. Links should be listed under the title: FIXED [:heavy_check_mark:]

| ACTUAL WEB PAGE | OURS |
| :-------------: | :---: |
| <img width="1440" alt="tnw-latest-funding" src="https://user-images.githubusercontent.com/62937819/90028310-1b803100-dcec-11ea-99e1-19b29c645133.png"> | <img width="1440" alt="our-latest-funding" src="https://user-images.githubusercontent.com/62937819/90028172-ebd12900-dceb-11ea-9c76-3976af86b5c8.png"> |

5. Align center elements in the footer. It contains social media icons vertically and horizontally aligned ([link to page](https://thenextweb.com/)): FIXED [:heavy_check_mark:] 


| ACTUAL WEB PAGE | OURS |
| :-------------: | :---: |
| ![image](https://user-images.githubusercontent.com/24902525/89915822-774b9c80-dbbc-11ea-82f4-18ceb9ab8876.png) | <img width="771" alt="footer-aligned" src="https://user-images.githubusercontent.com/62937819/90028166-ea9ffc00-dceb-11ea-90b7-965f1915018a.png"> |

6. No inline styling: FIXED [:heavy_check_mark:] 

The only inline style came about because we copied the W3C CSS Validator results (CSS level 3 + SVG) to show our readers that we've taken the care to create an interoperable Web page. In the code, we overlooked that it contained an inline style. We generally do not use inline styling.

7. Using the Same Rule Repetitively :pray:

Counter 1:

- We are aware that it's best to apply generic classes as much as possible. However, there are instances when applying a generic class repetitively actually means writing more code than with applying directly styling the element.

Counter 2: 

- When using pre-processors like SASS, the compiled CSS actually repeats a lot of code. The image below is a snippet of code from an approved capstone project which utilized SASS. As you can see, it has repeated code as a result of using SASS.

![image](https://user-images.githubusercontent.com/62937819/90034994-e11a9200-dcf3-11ea-92d2-46bbb80e80bb.png)
 
8. Using Inefficient CSS Selectors :pray:

Counter 1:  I understand that the inefficient use of CSS is not advised, but I don't think that is the case with our code here. Our approach is similar to pre-processors like SASS, which utilizes specific selector tags. 

Counter 2: The image below is from the same approved capstone project. If you look at line 473, this is what happens when you repeatedly assign a class to every element -- it becomes quite messy. Line 474 is actually the same code. As mentioned earlier, pre-processors like SASS actually compile CSS to be nested within each other multiple times, without adversely affecting the specificity of the code.

![image](https://user-images.githubusercontent.com/62937819/90035065-f68fbc00-dcf3-11ea-945f-eafb1cc49cab.png)

> What we hope ever to do with ease, we must learn first to do with diligence."
~Samuel Johnson

bertil291utn commented:
**Status:** Approved ✅

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.



<h2 align="center">Ship it </h2>

<p align="center"><img src="https://i.cubeupload.com/4HbSFK.jpg" alt="image" width="100" /></p>

### Summary

- Hey, guys thank you for the (large) explanation, I see you put all your effort on this project.

Please merge to master branch and move on with the next assignment.

You're doing  a great job keep going 💻 🚀 🔥

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”
